api-core/src/main/java/com/vimeo/networking2/internal/interceptor/UserAgentHeaderInterceptor.kt

```diff
@@ -32,8 +32,9 @@ import okhttp3.Response
  * @param userAgent The user agent that should be sent with every request.
  */
 class UserAgentHeaderInterceptor(private val userAgent: String) : Interceptor {
-    override fun intercept(chain: Interceptor.Chain): Response =
-        chain.proceed(chain.request().newBuilder().header(HEADER_USER_AGENT, userAgent).build())
+    override fun intercept(chain: Interceptor.Chain): Response = chain.proceed(chain.request().run {
+        newBuilder().headers(headers().newBuilder().addUnsafeNonAscii(HEADER_USER_AGENT, userAgent).build()).build()
+    })
```

Comment:
**Optional:** This feels more terse then concise to me, it too me a minute to understand what was happening with the `run` call here, I think it would be more readable broken up but not a blocker.

Reply:
I decided to use `let` with a named parameter and split out the header creation into a separate local property. I think it's a little more legible now.